apis/flinkcluster/v1beta1/flinkcluster_validate.go

```diff
@@ -231,7 +231,12 @@ func (v *Validator) checkSavepointGeneration(
 }
 
 func (v *Validator) validateTaskManagerUpdate(old *FlinkCluster, new *FlinkCluster) error {
-	if old.Spec.TaskManager.DeploymentType != new.Spec.TaskManager.DeploymentType {
+	//  When flink-operator updated, old flinkCluster does not have TaskManager.DeploymentType, may cause update failed.
+	if old.Spec.TaskManager.DeploymentType == "" && (new.Spec.TaskManager.DeploymentType != "" && new.Spec.TaskManager.DeploymentType != DeploymentTypeStatefulSet) {
```

Comment:
can we mode old/new DploymentTypes into `var`s I'm guessing it will ease up the reading here.